### DISCUSSION TITLE
( Help ) can't manipulate loaded buffer

### DISCUSSION THREAD

**Discussion by ahsanabbasi404** [katspaugh/wavesurfer.js#3812] [General]
Hello,

I hope this message finds you well. I have been using WaveSurfer for some time now and I have a question regarding real-time buffer manipulation.

Specifically, I would like to use Tone.js in conjunction with WaveSurfer, as they both point to the same audio source. My current setup allows me to control playback, stop, and seek functions using WaveSurfer, while Tone.js handles audio manipulations such as changing pitch, bass, and equalization. To achieve this, I load the audio file into both WaveSurfer and Tone.js, keeping WaveSurfer muted for visualization purposes only, displaying the elapsed audio time, while Tone js plays the audio.

While this method works, it feels somewhat cumbersome and I am concerned that it might lead to potential issues. I came across an example of using the Web Audio API with WaveSurfer, but it involved an audio element, which I found a bit confusing.

I have my own custom controllers for playing, stopping, and seeking playback. Is there a more streamlined solution for integrating WaveSurfer with Tone.js, allowing for real-time audio manipulation while maintaining WaveSurfer’s visualization capabilities?

**Comment by katspaugh** [comment]
You can refer to this example, it shows how to connect a Web Audio buffer from wavesurfer to external Web Audio nodes:

https://vocoder.wavesurfer-js.pages.dev/#phase-vocoder/index.js
